contracts/sdk/replication/RegistrationSMTReplicator.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.22;
+
+import {ERC1967Utils} from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Utils.sol";
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import {UUPSUpgradeable} from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+
+import {SetHelper} from "@solarity/solidity-lib/libs/arrays/SetHelper.sol";
+import {TypeCaster} from "@solarity/solidity-lib/libs/utils/TypeCaster.sol";
+import {AMultiOwnable} from "@solarity/solidity-lib/access/AMultiOwnable.sol";
+
+import {IPoseidonSMT} from "../../interfaces/state/IPoseidonSMT.sol";
+
+/**
+ * @title State Replicator for Registration SMT
+ * @notice This contract is meant to be deployed on the chain for the Registration SMT is not available.
+ * The owner of the replicator will be able to set a set of oracles that can transition the root.
+ */
+contract RegistrationSMTReplicator is IPoseidonSMT, AMultiOwnable, UUPSUpgradeable {
+    using TypeCaster for *;
+
+    using SetHelper for EnumerableSet.AddressSet;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    uint256 public constant ROOT_VALIDITY = 1 hours;
```

Comment:
Also make it configurable?

Reply:
Done, but there is a thing with backwards compatibility